lib/src/lib.rs

```diff
@@ -166,7 +166,8 @@ pub struct Request {
     pub id: Id,
     pub jsonrpc: Version,
     pub method: String,
-    pub params: Params,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Can we get the same result if we slap a similar annotation onto the inner variants of `Params`?

That would allow us to not break the public API and release this as a patch.

If we have to use an `Option`, I would prefer if we:

1. Initialize it to `None`
2. Set it to `Some` on the first call to `push`
3. Make `serialize` an immutable getter again

Reply:
Actually, we should only initialize `Params` to `None` for the `v1` request. It is only for `v2` where it can be omitted.

Reply:
So we can't use `serde skip_if` on enums with variants. In order to not break the `pub` on the struct, I have implemented `Serializer` to detect if map is empty.

Reply:
I have also updated change log and bumped version (since no public changes).